File: common/src/main/java/generations/gg/generations/core/generationscore/common/network/packets/S2CUnlockReloadPacket.java
package generations.gg.generations.core.generationscore.common.network.packets;

import net.minecraft.network.FriendlyByteBuf;
import net.minecraft.resources.ResourceLocation;
import org.jetbrains.annotations.NotNull;

import static generations.gg.generations.core.generationscore.common.GenerationsCore.id;

public class S2CUnlockReloadPacket implements GenerationsNetworkPacket<S2CUnlockReloadPacket> {
    public static ResourceLocation ID = id("unlock_reload");

    public static S2CUnlockReloadPacket decode(FriendlyByteBuf buffer) {
        return new S2CUnlockReloadPacket();
    }

    @Override
    public ResourceLocation getId() {
        return ID;
    }

    public void encode(@NotNull FriendlyByteBuf buffer) {}

}

File: common/src/main/java/generations/gg/generations/core/generationscore/network/S2CUnlockReloadPacket.kt
package generations.gg.generations.core.generationscore.network

import com.cobblemon.mod.common.api.net.ClientNetworkPacketHandler
import com.cobblemon.mod.common.api.net.NetworkPacket
import generations.gg.generations.core.generationscore.GenerationsCore.id
import generations.gg.generations.core.generationscore.GenerationsDataProvider
import net.minecraft.client.Minecraft
import net.minecraft.network.FriendlyByteBuf

class S2CUnlockReloadPacket : NetworkPacket<S2CUnlockReloadPacket> {
    companion object {
        val ID = id("unlock_reload")

        fun decode(buffer: FriendlyByteBuf) = S2CUnlockReloadPacket()
    }

    override val id = ID
    override fun encode(buffer: FriendlyByteBuf) {}

     object UnlockReloadPacketHandler : ClientNetworkPacketHandler<S2CUnlockReloadPacket> {
         override fun handle(packet: S2CUnlockReloadPacket, client: Minecraft) {
             GenerationsDataProvider.canReload = true
         }
     }
}